### PR TITLE
[iOS] Re-initialize iOS App

### LINKED ISSUE #441: [iOS] Re-initialize iOS Project
## Kind (Required)

- Improvement

 ## Overview (Required)

- Re-initialize iOS Project to satisfy following requirements.
  - Use "Swift Package Manager Project Definition".
    - If this project should divide configuration(such as `Debug` and `Production` ), please divide Xcode Project.

 ## Links	

- https://github.com/DroidKaigi/conference-app-2021/issues/439#issuecomment-848549228
- [Swift PMとマルチプロジェクトでBuild Configurationをクリーンに保つ](https://www.notion.so/date/Swift-PM-Build-Configuration-4f14ceac795a4338a5a44748adfeaa40)

### PR DESCRIPTION
**Pull request by ry-itto**
## Issue
- close #441 

## Overview (Required)
- Re-initialize iOS App Project
  - Run App Configurations with divided Xcode Projects
- Setup project based on https://github.com/DroidKaigi/conference-app-2021/issues/439
- Add document to README.md
- There are no implementations about features.
- No change to build tools, but this is no reason.
  - If it's necessary, switch to manage build tools with `SPM`.

### TODOs
- [x] Initialize AppScreen with 4 tabs

## Links
-

## Screenshot
Before | After
:--: | :--:
 | スクリーンショット 2021-05-28 23 47 23

### CODE REVIEW

**Comment by ry-itto** [comment]
@takahirom 
Please review this PR 🙏🏼 

After this PR merged, we are ready to implement iOS App features.

**Comment by takahirom** [comment]
:eyes:

File: ios/.gitignore
Comment (ostk0069) [inline ios/.gitignore:4]: maybe `ios/DroidKaigi 2021/Debug.xcodeproj` &  `ios/DroidKaigi 2021/Release.xcodeproj` is not ignored ?
Or is this line not needed?

File: ios/Makefile
Comment (ostk0069) [inline ios/Makefile:5]: 😎

File: ios/Sources/AppFeature/AppScreen.swift
Comment (ostk0069) [inline ios/Sources/AppFeature/AppScreen.swift:20]: It seems better to use enum & ForEach instead of hard code tag number.

File: ios/Makefile
Comment (ostk0069) [inline ios/Makefile:9]: I though using SPM for manage XcodeGen is better. I want to know your idea using mint.

File: ios/Sources/AppFeature/AppScreen.swift
Comment (ostk0069) [inline ios/Sources/AppFeature/AppScreen.swift:16]: It seems better to use any router or coordinator classes to call views.

**Review by ostk0069 — COMMENTED** [review]
I reviewed this PR because @ry-itto requested me to do so. Thanks for requesting me! 👍 
Good luck with iOS App leader!

File: ios/Makefile
Comment (ry-itto) [reply ios/Makefile:9]: `XcodeGen` means `SwiftLint` ..? 🤔 

No reasons for using mint in this PR 😢  
I will fix with using SPM for managing tools.

File: ios/.gitignore
Comment (ry-itto) [reply ios/.gitignore:4]: > ios/DroidKaigi 2021/Debug.xcodeproj & ios/DroidKaigi 2021/Release.xcodeproj is not ignored ?

Yes. these are ignored.
This ignore line is for Root `*.xcodeproj`  such as `DroidKaigi2021.xcodeproj` generated by `swift package generate-xcodeproj` command.

**Review by takahirom — APPROVED** [review]
Thank you. Basically, I leave it to @ry-itto 🙏 
Please tell me if you should ask for the help of an iOS engineer on Twitter etc. 🙏

File: ios/Sources/AppFeature/AppScreen.swift
Comment (ry-itto) [reply ios/Sources/AppFeature/AppScreen.swift:16]: It's good!
However I think this is overdoing in this PR. So I will make other GitHub issues. 🙏🏼

**Comment by ry-itto** [comment]
@ostk0069 
I fixed! Please re-review 🙏🏼

File: ios/Makefile
Comment (ostk0069) [inline ios/Makefile:18]: I don't think this command is needed. This command is used only when setup configuration which is done.

File: ios/swiftgen.yml
Comment (ostk0069) [inline ios/swiftgen.yml:4]: I want to know the purpose for this template. If it's okay, let me know.

And also, if you're using SwiftGen, why don't you use for strings as well?

**Review by ostk0069 — COMMENTED** [review]
Thanks for creating issues! 

Maybe you should update the description of this PR.
I left few comments!

File: ios/Makefile
Comment (ry-itto) [reply ios/Makefile:18]: I left this command just in case. For example, in case  adding new Localizable.strings file.
I'm thinking of doing tasks for localization ​​with other PR. 🙏🏼

File: ios/swiftgen.yml
Comment (ry-itto) [reply ios/swiftgen.yml:4]: In SPM Tool management, using SwiftGen has an issue.

`swift run -c release --package-path Tools swiftgen` : fails
`mint run swiftgen` : success

And I tried to use with `templatePath` instead of `templateName` , it succeeded.
So I use `templatePath` in this PR.

I think this is related to this forum.
https://github.com/apple/swift-evolution/blob/main/proposals/0303-swiftpm-extensible-build-tools.md

**Review by ostk0069 — APPROVED** [review]

**Comment by ry-itto** [comment]
Merge this 🙏🏼

**Pull request merged**
